modules/navier_stokes/include/materials/NSFVFrictionFlowDiodeMaterial.h

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FunctorMaterial.h"
+
+/**
+ * Additional anistropic friction linear or quadratic terms
+ */
+class NSFVFrictionFlowDiodeMaterial : public FunctorMaterial
+{
+public:
+  static InputParameters validParams();
+
+  NSFVFrictionFlowDiodeMaterial(const InputParameters & parameters);
+
+private:
+  /// Direction of the diode
+  const RealVectorValue & _direction;
+
+  /// Magnitude of the additional linear resistance
+  const RealVectorValue & _linear_resistance;
+
+  /// Magnitude of the additional quadratic resistance
+  const RealVectorValue & _quadratic_resistance;
```

Comment:
```suggestion
  const RealVectorValue _direction;

  /// Magnitude of the additional linear resistance
  const RealVectorValue _linear_resistance;

  /// Magnitude of the additional quadratic resistance
  const RealVectorValue _quadratic_resistance;
```
Avoids querying the pointer to the memory location (not that this is likely to be important)